[package] apr/1.7.0: consumer packages fail to find apr when used as shared

### Package and Environment Details
  * Package Name/Version: **apr/1.7.0**
  * Operating System+version: **Windows 10**
  * Compiler+version: **MSVC v142**
  * Conan version: **conan 1.35.0**
  * Python version: **Python 3.7.9**


### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Windows
os_build=Windows
arch=x86
arch_build=x86_64
compiler=Visual Studio
compiler.version=16
build_type=Release
[options]
[build_requires]
[env]
```


### Steps to reproduce (Include if Applicable)
Install apr/1.7.0 as shared
conan install apr/1.7.0@ -o shared=True --build=apr

Try installing a consumer of apr. Trying with serf
conan install serf/1.3.9@ -o apr:shared=True -o apr-util:shared=True --build=serf --build=apr-util

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
LINK : fatal error LNK1181: cannot open input file 'libapr-1.lib'
scons: *** [libserf-1.dll] Error 1181
scons: building terminated because of errors.
serf/1.3.9:
serf/1.3.9: ERROR: Package 'e896ac717b33682561d2e328d65b7fe6a0bec462' build failed
serf/1.3.9: WARN: Build folder C:\projects\.conan\data\serf\1.3.9\_\_\build\e896ac717b33682561d2e328d65b7fe6a0bec462
ERROR: serf/1.3.9: Error in build() method, line 111
        self.run("scons {} {}".format(" ".join(escape_str(s) for s in args), " ".join("{}={}".format(k, escape_str(v)) for k, v in kwargs.items())), run_environment=True)
```

</details>
Patch 0001 changes the name of the library binary from libapr-1 to apr-1 when compiled as shared. This was done so conan only compiles one library, either static or shared, not both as apr usually does.

This name change affects compilation of consumer packages like serf which expects the libapr-1 name when linking the library as shared.

I propose to make what apr-util does, droping the name change and make cpp_info.libs take the correct name
https://github.com/cesenaLA/conan-center-index/commit/bbc084c0f09f09070ed840eb3c8239d0d9d1b152

--- a/recipes/apr/all/conanfile.py
+++ b/recipes/apr/all/conanfile.py
@@ -120,7 +120,7 @@ class AprConan(ConanFile):
 
     def package_info(self):
         self.cpp_info.names["pkg_config"] = "apr-1"
-        self.cpp_info.libs = ["apr-1"]
+        self.cpp_info.libs = ["libapr-1" if self.settings.compiler == "Visual Studio" and self.options.shared else "apr-1"]
         if not self.options.shared:
             self.cpp_info.defines = ["APR_DECLARE_STATIC"]
             if self.settings.os == "Linux":

--- a/recipes/apr/all/patches/0001-cmake-build-only-shared-static.patch
+++ b/recipes/apr/all/patches/0001-cmake-build-only-shared-static.patch
@@ -1,6 +1,6 @@
 --- CMakeLists.txt
 +++ CMakeLists.txt
-@@ -258,27 +258,27 @@
+@@ -258,27 +258,26 @@
  SET(install_targets)
  SET(install_bin_pdb)
  
@@ -11,7 +11,6 @@
  SET(install_bin_pdb ${install_bin_pdb} ${PROJECT_BINARY_DIR}/libapr-1.pdb)
  TARGET_LINK_LIBRARIES(libapr-1 ${APR_SYSTEM_LIBS})
  SET_TARGET_PROPERTIES(libapr-1 PROPERTIES COMPILE_DEFINITIONS "APR_DECLARE_EXPORT;WINNT")
-+SET_TARGET_PROPERTIES(libapr-1 PROPERTIES PREFIX "" OUTPUT_NAME "apr-1")
  ADD_DEPENDENCIES(libapr-1 test_char_header)
 -
 +ELSE()

